### PR TITLE
seperate bazel build and test for PRs

### PR DESCRIPTION
**Pull request by BenTheElder**
this matters now since we are running jobs _after_ the bazel build is complete.

### CODE REVIEW

**Comment by krzyzacy** [comment]
consider make two none-blocking jobs first, then migrate the existing bazel job over... This has a non-zero chance of blocking sq.

File: mungegithub/misc-mungers/deployment/kubernetes/configmap.yaml [hunk 1]
```diff
@@ -16,7 +16,8 @@ github-key-file: /etc/hook-secret-volume/secret
 # !!! This *MUST* be kept in sync with
 #     submit-queue/deployment/kubernetes/configmap.yaml !!!
 required-retest-contexts: "\
-  pull-kubernetes-bazel,\
+  pull-kubernetes-bazel-build,\
```
Comment (ixdy) [inline mungegithub/misc-mungers/deployment/kubernetes/configmap.yaml]: did you update the SQ?

Comment (ixdy) [reply mungegithub/misc-mungers/deployment/kubernetes/configmap.yaml]: hm, seems like `mungegithub/submit-queue/deployment/kubernetes/configmap.yaml` wasn't updated?

Comment (krzyzacy) [reply mungegithub/misc-mungers/deployment/kubernetes/configmap.yaml]: oh oh oh oh oh - too many places

and does sq picks up config change automatically now or we need to redeploy that

Comment (ixdy) [reply mungegithub/misc-mungers/deployment/kubernetes/configmap.yaml]: I imagine redeploy is needed, but I'm not sure.

Comment (BenTheElder) [reply mungegithub/misc-mungers/deployment/kubernetes/configmap.yaml]: config has been updated, I asked @cjwagner about redeploying.

**Pull request merged**